src/stream/ws/handshake.c

```diff
@@ -70,7 +72,7 @@ ssize_t ws_handshake(char* output, size_t output_maxlen, const char* input)
 	bool have_protocols = strlen(protocols) != 1;
 	bool have_versions = strlen(versions) != 1;
 
-	if (have_protocols && !strstr(protocols, ",chat,"))
+	if (have_protocols && !strstr(protocols, ",binary,"))
```

Comment:
It looks like websockify also uses "binary" as a subprotocol, so this change should be OK.

However, I'm not really sure why I added this in the first place as both firefox and chromium on my system work fine with this as is. They don't specify subprotocols, so they don't run into this.

Which version of firefox are you running?

Reply:
Ok I understand what is going on. It is not the browser defining the subprotocols, but they are passed as optional arguments to the websocket constructor. Since I use an ancient version of noVNC, it passes the subprotocols binary and base64. Firefox will then send the header 'Sec-WebSocket-Protocol: binary, base64'. note, with a space. NoVNC dropped the base64 support and later removed defaulting to binary, see [1], which explains why you don't have the header at all.

[1] https://github.com/novnc/noVNC/commit/c912230309806aacbae4295faf7ad6406da97617

Reply:
It's tempting to remove this stuff entirely and require newer noVNC and browser, but at least it's correct with your change so we'll start with that.

Reply:
I leave it up to you. I updated the commit message accordingly.